consumer.go

```diff
@@ -212,3 +272,17 @@ func (c *consumer) loop() {
 		}
 	}
 }
+
+func (c *consumer) newIncrementalReader() (*Reader, []*bfs.Object, error) {
+	manifest, err := loadManifest(c.ctx, c.remote)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	files := manifest.Files
+	remotes := make([]*bfs.Object, 0, len(files))
+	for _, file := range files {
+		remotes = append(remotes, bfs.NewObjectFromBucket(c.bucket, file))
+	}
+	return MultiReader(c.ctx, remotes, &c.opt.ReaderOptions), remotes, nil
```

Comment:
yes, this is a bit "meh". I suggest you do something like:

```
reader := MultiReader(...)
reader.ownRemotes = true
return reader, nil
```

then in `Reader.Close` you know to close `remotes` too if `ownRemotes` is set.